File: s05_OOP/OOPDetails/app/src/main/java/com/omeryildizce/oopdetails/MainActivity.java
package com.omeryildizce.oopdetails;


import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Kedi kedi1 = new Kedi("kedi","sarı", "beyaz");
        Kedi kedi2 = new Kedi();
        // turIsmi = "test";
        // System.out.println(turIsmi);
        System.out.println(Kedi.turIsmi);
        kedi1.konusedi();
        Kedi.konusediStatic();
    }
}

File: s29_AdvancedFunction/Delegates/app/src/main/java/com/omeryildizce/delegates/MainActivity.kt
package com.omeryildizce.delegates

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.Lifecycle.Event.*
import androidx.lifecycle.LifecycleEventObserver
import androidx.lifecycle.LifecycleOwner

class MainActivity : AppCompatActivity() , LifecycleLogger by LifecycleLoggerImplementation(){
    // property delegate
    private  val myVariable by lazy {
        println("hello  this is a layz implementation")
        10  }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        registerLifecycleOwner(this)
        println(myVariable)
    }

    override fun onResume() {
        super.onResume()
    }


}


interface LifecycleLogger{
    fun registerLifecycleOwner(owner:LifecycleOwner)
}

class LifecycleLoggerImplementation : LifecycleLogger, LifecycleEventObserver{
    override fun registerLifecycleOwner(owner: LifecycleOwner) {

        owner.lifecycle.addObserver(this)
    }

    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
        when(event){
            ON_RESUME ->  println("onresume")
            ON_PAUSE ->  println("onpause")
            ON_DESTROY ->  println("ondestroy")
            ON_CREATE ->  println("oncreate")
            ON_START -> println("onstart ")
            ON_STOP -> println("onstop")
            ON_ANY -> println("onany")
        }

    }
}